src/kagglehub/models.py

```diff
@@ -13,22 +14,44 @@
 DEFAULT_IGNORE_PATTERNS = [".git/", "*/.git/", ".cache/", ".huggingface/"]
 
 
-def model_download(handle: str, path: Optional[str] = None, *, force_download: Optional[bool] = False) -> str:
+def model_download(
+    handle: str,
+    path: Optional[str] = None,
+    *,
+    force_download: Optional[bool] = False,
+    output_dir: Optional[str] = None) -> str:
     """Download model files.
 
     Args:
         handle: (string) the model handle.
         path: (string) Optional path to a file within the model bundle.
         force_download: (bool) Optional flag to force download a model, even if it's cached.
-
+        output_dir: (str) Optional path to copy model files to after successful download.
 
     Returns:
         A string representing the path to the requested model files.
     """
     h = parse_model_handle(handle)
     logger.info(f"Downloading Model: {h.to_url()} ...", extra={**EXTRA_CONSOLE_BLOCK})
-    return registry.model_resolver(h, path, force_download=force_download)
+    cached_dir = registry.model_resolver(h, path, force_download=force_download)
+
+    if output_dir is None:
```

Comment:
we dont need to copy if the cached_dir is the same as output_dir

Reply:
added an extra equality check for this